### PR TITLE
JAVA-1476: Improve GraphSON2 ser/de performance.

### PR DESCRIPTION
**Pull request by newkek**
Largely rewrote the Gremlin structure objects deserializers to deserialize GraphSON2 into our driver own Graph structure objects like Vertex, Edge, etc..

The goal is to avoid the abstract map creation which reads all fields in the typed element, and exposes them to the abstraction I had written for the deserializers. Turns out reading in an abstract way like this causes Jackson to execute a lot of reflection calls internally, which generally slows down the deserialization process. The solution to that is to rewrite the deserializers in a slightly more strict way and make use of the JsonParser API, to deserialize the JSON payload token by token.

Performance benchmarks show a 3x throughput increase with these changes:
```
Deserializing a list of 20 000 vertices in GraphSON2 (multiple runs' times in nanoseconds): 

Without java1476 
3792446372
4388204010
4508486046
4297619753
4496564199

With java1476:
1477106395
1999709701
1912950242
1571789246
1563224421
```

### CODE REVIEW

File: driver-core/src/main/java/com/datastax/driver/dse/graph/GraphSONTokens.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public static final String KEY = "key";
+    public static final String IN = "inV";
+    public static final String OUT = "outV";
+    public static final String LABEL = "label";
+    public static final String LABELS = "labels";
+    public static final String OBJECTS = "objects";
+    public static final String IN_LABEL = "inVLabel";
+    public static final String OUT_LABEL = "outVLabel";
+
+    public static final String BULK = "bulk";
```
Comment (newkek) [inline driver-core/src/main/java/com/datastax/driver/dse/graph/GraphSONTokens.java]: going to make use of that in #75

**Comment by newkek** [comment]
Also corrected a potential NPE in `toString()` for Properties and VertexProperties that don't have a parent. (e.g. a gremlin traversal like `g.V().properties()`, we only get properties and can't tie them to a parent)

**Comment by tolbertam** [comment]
looks like mvn build is failing because GraphSONTokens.java is missing license header.

**Comment by newkek** [comment]
The 5.1 jenkins build passed and the 5.0 didn't but it doesn't seem related at all with this part of the code.

**Pull request merged**